src/Aspire.Hosting/ApplicationModel/ContainerImage.cs

```diff
@@ -0,0 +1,21 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// Represents the fully‑qualified container image reference that should be deployed.
+/// </summary>
+[DebuggerDisplay("{ValueExpression}")]
+public readonly record struct ContainerImage(IResource resource) : IManifestExpressionProvider
```

Comment:
Should these types have `Expression` in the name? `ContainerImageExpression`, `ContainerPortExpression`?

Reply:
I don't see us suffixing other implementors of `IManifestExpressionProvider` with `Expression` so maybe fine to keep as is. Also, I think not having "Expression" does a better job of highlighting that this is meant to be a semantic concept?

Reply:
Not sure about expression but it does feel like the name is too general purpose. Feels like a "parameter" more than an expression.

Reply:
Yeah, this isn't a `ContainerImage`, it is a pointer to a `ContainerImage`.